lib/Extended_ast.ml

```diff
@@ -101,6 +107,26 @@ module Parse = struct
 
   let normalize fg x = map fg fix_letop_locs @@ map fg normalize_lists @@ x
 
+  let repl_file (lx : Lexing.lexbuf) =
+    let str = String.strip (Bytes.to_string lx.lex_buffer) in
+    let phrases = Astring.String.cuts ~sep:"\n# " str in
+    let phrases =
+      List.mapi phrases ~f:(fun i p ->
+          let p = String.strip p in
+          match i with
+          | 0 -> Option.value (String.chop_prefix p ~prefix:"# ") ~default:p
+          | _ -> p )
+    in
+    List.map phrases ~f:(fun p ->
+        let p, prepl_output =
+          match Astring.String.cut ~sep:";;\n" p with
```

Comment:
Hmm maybe we could print a warning in such a case? I'm not sure making the parsing more complex to handled old versions is the way to go, we should encourage users to make their code "nice"

Reply:
I'm not a fan of warnings. I'm not sure that would make the parsing more complex than the current.

Reply:
I modified this function to raise a `Syntaxerr.Error Expecting` exception, to use the same interface as the existing parsing functions from `Parse`, otherwise we would need to wrap a result type around everything, and we would just print the input verbatim and not display any error/warning anyway.